packages/core/src/extension/extension-iframe.ts

```diff
@@ -18,6 +18,14 @@ export class ExtensionIframe {
     }
 
     async attach(): Promise<void> {
+        if (
+            document
+                .querySelector(`[data-extension-id="${this._extension.id}"]`)
+                ?.querySelector('iframe')
+        ) {
+            return;
+        }
+
```

Comment:
During the UI implementation, I noticed that sometimes the iframe gets reattached when the React state updates. Therefore, adding a check here to ensure that only one iframe attached for an extension.